himan-scripts/bl-turbulence.lua

```diff
@@ -19,24 +19,48 @@
 
 logger:Info("Calculating boundary layer turbulence")
 
+-- fetch producer information
+local currentProducer = configuration:GetTargetProducer()
+local currentProducerName = currentProducer.GetName(currentProducer)
+
 local turbparam = param("BLTURB-N")
 
 -- Surface (10m) wind speed in m/s = lowest model level = 65
+local lev = {}
 local ws = param("FF-MS")
-local lev = level(HPLevelType.kHybrid, 65)
+if currentProducerName == "ECGMTA" then
+  lev = level(HPLevelType.kHeight, 10)
+else
+  lev = level(HPLevelType.kHybrid, 65)
+end
 
 local wsms = luatool:FetchWithType(current_time, lev, ws, current_forecast_type)
 
-if not wsms then
-  logger:Error("No data found")
-  return
+-- Wind gust
+local gustlevel = {}
+local gust = param("FFG-MS")
+if currentProducerName == "ECGMTA" then
+  gustlevel = level(HPLevelType.kGround, 0)
```

Comment:
Just did this:

```
radon=> insert into param_level_transform values (default, 131, (select id from param where name = 'FFG-MS'), 1, 0, 6, 10);
INSERT 0 1
```

So Himan automatically knows to fetch FFG-MS from ground/0 if it's requested for height/10.